R/dummy_data.R

```diff
@@ -649,3 +654,317 @@ dummy_admdad <- function(id, admtime) {
   )
   return(res)
 }
+
+#' @title
+#' Generate simulated locality variables data
+#'
+#' @description
+#' This function creates a dummy dataset with a subset of variables that
+#' are contained in the GEMINI "locality-variables" table, as seen in
+#' [GEMINI Data Repository Dictionary](https://geminimedicine.ca/the-gemini-database/).
+#'
+#' This data table will simulate dissemination area IDs based on Canadian census data.
+#'
+#' @param dbcon (`DBIConnection`)\cr
+#'  A connection to the GEMINI database, used to access the 2021 Canadian census dissemination codes.
+#' It is required when `da21uid` is missing.
+#'
+#' @param nid (`integer`)\cr Number of unique encounter IDs to simulate. In this data table, each ID occurs once.
+#' It is optional when `cohort` is provided.
+#'
+#' @param n_hospitals (`integer`)\cr Number of hospitals in simulated dataset.
+#' It is optional when `cohort` is provided.
+#'
+#' @param da21uid (`integer` | `vector`)\cr Allows the user to customize which location ID to include in the output.
+#' It is required when `dbcon` is missing. It can be an integer or an integer vector.
+#'
+#' @param seed (`integer`)\cr Optional, a number to be used to set the seed for reproducible results.
+#'
+#' @param cohort (`data.frame|data.table`) Optional, an existing data frame or data table similar to `admdad` in
+#' GEMINI with at least the following columns:
+#' - `genc_id` (`integer`): GEMINI encounter ID
+#' - `hospital_num` (`integer`): Hospital ID
+#' If `cohort` is provided, `nid` and `n_hospital` inputs are not used.
+#'
+#' @return (`data.table`)\cr
+#' A data.table object similar to the "locality_variables" table that contains the following fields:
+#' - `genc_id` (`integer`): GEMINI encounter ID
+#' - `da21uid` (`integer`): Dissemination area ID based on 2021 Canadian census data using PCCF Version 8A
+#'
+#' @export
+
+dummy_locality <- function(dbcon = NULL, nid = 1000, n_hospitals = 10, cohort = NULL, da21uid = NULL, seed = NULL) {
+  ### check inputs
+  if (is.null(dbcon) && is.null(da21uid)) {
+    stop("A DB connection or list of dissemination codes is required.")
+  } else if (!is.null(da21uid)) {
+    check_input(da21uid, "integer")
+  } else {
+    check_input(dbcon, c("DBI", "dbcon", "PostgreSQL"))
+  }
```

Comment:
the last statement needs to be conditioned on non-null dbcon, and the input checks could be cleaner (no need to use if-else):

```suggestion
  ### check inputs
  if (is.null(dbcon) && is.null(da21uid)) {
    stop("A DB connection or list of dissemination codes is required.")
  }  
  
  if (!is.null(da21uid)) {
    check_input(da21uid, "integer")
  } 
  
  if (!is.null(dbcon)) {
    check_input(dbcon, c("DBI", "dbcon", "PostgreSQL"))
  }
```